src/main/java/com/qunite/api/security/JwtService.java

```diff
@@ -3,32 +3,78 @@
 import com.auth0.jwt.JWT;
 import com.auth0.jwt.algorithms.Algorithm;
 import com.auth0.jwt.interfaces.DecodedJWT;
+import com.qunite.api.data.UserRepository;
+import com.qunite.api.domain.Tokens;
 import com.qunite.api.domain.User;
+import com.qunite.api.service.TokenService;
+import com.qunite.api.web.dto.auth.AuthenticationResponse;
 import java.time.Instant;
+import java.time.temporal.ChronoUnit;
 import java.util.Optional;
+import lombok.RequiredArgsConstructor;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
-@Component
+@Service
+@RequiredArgsConstructor
 public class JwtService {
   @Value("${jwt.secret}")
   private String secret;
 
   @Value("${spring.application.name}")
   private String issuer;
 
-  public String createJwtToken(User user) {
+  private final TokenService tokenService;
+
+  private final UserRepository userRepository;
+
+  @Transactional
```

Comment:
Why `@Transactional`

Reply:
because this method saves token to repository and all service methods at our project have transactional on em-_-

Reply:
well, repo methods has its own transactional envitonment, so we don`t need to use transactional at all in such situations, but ok, `ll delete it.